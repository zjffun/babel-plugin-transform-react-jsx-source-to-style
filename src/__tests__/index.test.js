const babel = require("@babel/core");
const plugin = require("../../");

const babelConfig = {
  filename: "/fake/path/mock.js",
  presets: ["@babel/preset-react"],
  plugins: [plugin],
};

it("basic-sample", () => {
  const example = `
var x = <sometag />;
`;
  const { code } = babel.transform(example, babelConfig);
  expect(code).toMatchSnapshot();
});

it("no-jsx", () => {
  const example = `
var x = 42;
`;
  const { code } = babel.transform(example, babelConfig);
  expect(code).toMatchSnapshot();
});

it("with-style add assign object style", () => {
  const example = `
var x = <sometag style={null} />;
var x = <sometag style={undefined} />;
var x = <sometag style={{'--source-code-location': test}} />;
var x = <sometag style={styleObj} />;
`;
  const { code } = babel.transform(example, babelConfig);
  expect(code).toMatchSnapshot();
});

it("with-style add add string style", () => {
  const example = `
var x = <sometag style="" />;
var x = <sometag style={""} />;
var x = <sometag style={0} />;
var x = <sometag style={1} />;
var x = <sometag style={"--source-code-location: test"} />;
`;
  const { code } = babel.transform(example, babelConfig);
  expect(code).toMatchSnapshot();
});
