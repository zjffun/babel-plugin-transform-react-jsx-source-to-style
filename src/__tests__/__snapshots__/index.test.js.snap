// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic-sample 1`] = `
"var _jsxFileName = "/fake/path/mock.js";
var x = /*#__PURE__*/React.createElement("sometag", {
  style: {
    '--source-code-location': _jsxFileName + ':' + 2 + ':' + 9
  }
});"
`;

exports[`no-jsx 1`] = `"var x = 42;"`;

exports[`with-style add add string style 1`] = `
"var _jsxFileName = "/fake/path/mock.js";
var x = /*#__PURE__*/React.createElement("sometag", {
  style: '--source-code-location:' + (_jsxFileName + ':' + 2 + ':' + 9) + ';' + ""
});
var x = /*#__PURE__*/React.createElement("sometag", {
  style: '--source-code-location:' + (_jsxFileName + ':' + 3 + ':' + 9) + ';' + ""
});
var x = /*#__PURE__*/React.createElement("sometag", {
  style: '--source-code-location:' + (_jsxFileName + ':' + 4 + ':' + 9) + ';' + 0
});
var x = /*#__PURE__*/React.createElement("sometag", {
  style: '--source-code-location:' + (_jsxFileName + ':' + 5 + ':' + 9) + ';' + 1
});
var x = /*#__PURE__*/React.createElement("sometag", {
  style: '--source-code-location:' + (_jsxFileName + ':' + 6 + ':' + 9) + ';' + "--source-code-location: test"
});"
`;

exports[`with-style add assign object style 1`] = `
"var _jsxFileName = "/fake/path/mock.js";
var x = /*#__PURE__*/React.createElement("sometag", {
  style: Object.assign({
    '--source-code-location': _jsxFileName + ':' + 2 + ':' + 9
  }, null)
});
var x = /*#__PURE__*/React.createElement("sometag", {
  style: Object.assign({
    '--source-code-location': _jsxFileName + ':' + 3 + ':' + 9
  }, undefined)
});
var x = /*#__PURE__*/React.createElement("sometag", {
  style: Object.assign({
    '--source-code-location': _jsxFileName + ':' + 4 + ':' + 9
  }, {
    '--source-code-location': test
  })
});
var x = /*#__PURE__*/React.createElement("sometag", {
  style: Object.assign({
    '--source-code-location': _jsxFileName + ':' + 5 + ':' + 9
  }, styleObj)
});"
`;
